src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentInvoker.cs

```diff
@@ -55,60 +65,33 @@ public class DefaultViewComponentInvoker : IViewComponentInvoker
             _logger = logger;
         }
 
-        public void Invoke(ViewComponentContext context)
+        /// <inheritdoc />
+        public async Task InvokeAsync(ViewComponentContext context)
         {
             if (context == null)
             {
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var method = ViewComponentMethodSelector.FindSyncMethod(
-                context.ViewComponentDescriptor.Type.GetTypeInfo(),
-                context.Arguments);
-            if (method == null)
+            var methodInfo = context.ViewComponentDescriptor?.MethodInfo;
```

Comment:
It's settable and we only verify if the context is non null at the start of the method.